sql/sql_expression.cc

```diff
@@ -1562,6 +1549,22 @@ parse(ML::Parse_Context & context, int currentPrecedence, bool allowUtf8)
             }
         }
 
+        if (context.match_literal('['))
```

Comment:
We want this to bind extremely tightly, so I would put this at the top of the while(true) loop, ie before the IS NULL part.